omod/src/main/java/org/openmrs/module/spa/servlet/SpaServlet.java

```diff
@@ -17,13 +20,23 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
+import static org.openmrs.module.spa.filter.SpaFilter.DEFAULT_SPA_BASE_URL;
+import static org.openmrs.module.spa.filter.SpaFilter.GP_KEY_SPA_BASE_URL;
+
 public class SpaServlet extends HttpServlet {
 
 	private static final long serialVersionUID = 1L;
 
 	@Override
 	protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
 		RequestDispatcher dispatcher = req.getRequestDispatcher("/module/spa/master-single-page-application.form");
+		req.setAttribute("openmrsBaseUrlContext", new String(req.getContextPath()));
+		try {
+			req.setAttribute("spaBaseUrlContext", Context.getAdministrationService().getGlobalProperty(GP_KEY_SPA_BASE_URL,
+					DEFAULT_SPA_BASE_URL));
+		} catch (APIException e) {
+			e.printStackTrace();
```

Comment:
My suspicion is that there is something better that we can do besides just printing the stacktrace. Maybe using a logger and logging the error. Also returning HTTP 500 from this servlet instead of a 200 that has an improper HTML file.

@dkayiwa can you comment here on best way to log the error and return an HTTP 500 from the servlet?

Reply:
https://wiki.openmrs.org/display/docs/Java+Conventions#JavaConventions-Donotprintout